pyproject.toml

```diff
@@ -3,12 +3,26 @@ addopts = "-rs -v"
 log_cli = true
 
 [tool.pyright]
-typeCheckingMode = "off"
-reportMissingTypeStubs = "error"
+typeCheckingMode = "strict"
 include = [
     "opentelemetry-api/src",
     "opentelemetry-sdk/src",
     "opentelemetry-semantic-conventions/src",
+    "opentelemetry-proto/src",
+    "scripts"
+]
+ignore = [
+  "docs/**",
+  "exporter/**",
+  "propagator/**",
+  "shim/**",
+]
+reportUnnecessaryTypeIgnoreComment = true
```

Comment:
One thing pattern I've noticed in contrib. If you do `type: ignore [foo]`, pyright seems to ignore everything. It only seems to narrow to the specific error if you use `pyright: ignore [foo]`.

I'm guessing there are quite a few places this is the case in the code today